complex/ft_test.c

```
@@ -310,6 +312,25 @@ void ft_next_iov_cnt(struct ft_xcontrol *ctrl, size_t max_iov_cnt)
 		ctrl->iov_iter = 0;
 }
 
+int ft_next_ctx(struct ft_xcontrol *ctrl, uint64_t comp)
+{
+	int ret;
+
+	ctrl->curr_ctx++;
+	if (ctrl->curr_ctx < ctrl->max_credits)
+		return 0;
+
+	while (ctrl->credits < ctrl->max_credits && comp == FI_SEND) {
+		ret = ft_comp_tx(FT_COMP_TO);
+		if (ret)
+			return ret;
+	}
+
+	ctrl->curr_ctx = 0;
+
+	return 0;
+}
```

I changed my mind.  If we're reading the CQ, then we need to propagate any failures back through the stack, so we can abort the test.